src/query/storages/fuse/src/statistics/column_statistic.rs

```diff
@@ -162,31 +182,35 @@ pub mod traverse {
     use databend_common_expression::types::DataType;
     use databend_common_expression::BlockEntry;
     use databend_common_expression::Column;
+    use databend_common_expression::ColumnBuilder;
+    use databend_common_expression::Value;
 
     use super::*;
 
-    pub type TraverseResult = Result<Vec<(Option<usize>, Column, DataType)>>;
+    pub type TraverseResult = Result<Vec<(Option<usize>, Value<AnyType>, DataType)>>;
 
     // traverses columns and collects the leaves in depth first manner
-    pub fn traverse_columns_dfs(columns: &[BlockEntry]) -> TraverseResult {
+    pub fn traverse_values_dfs(columns: &[BlockEntry]) -> TraverseResult {
         let mut leaves = vec![];
         for (idx, entry) in columns.iter().enumerate() {
             let data_type = &entry.data_type;
-            let column = entry.value.as_column().unwrap();
-            traverse_recursive(Some(idx), column, data_type, &mut leaves)?;
+            match &entry.value {
+                Value::Scalar(s) => {
+                    traverse_scalar_recursive(Some(idx), s, data_type, &mut leaves)?;
+                }
+                Value::Column(c) => {
+                    traverse_column_recursive(Some(idx), c, data_type, &mut leaves)?;
```

Comment:
It looks like `traverse_scalar_recursive` and `traverse_column_recursive` have a similar code structure, we can merge them into one function using `Value` as the argument and return type.

Reply:
not all the same,  let me try to see if it is better.

Reply:
I prefer current way, which is more clear to me.

traverse_column_recursive never call traverse_scalar_recursive
traverse_scalar_recursive call traverse_column_recursive some times

otherwise we still need to match Value in the unified func.